### PR TITLE
[cleanup] Delete PartitionKeysTimeWindowPartitionsSubset

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

This was intended as a perf optimization in the distant past, but improvements to common operations in the TimeWindowPartitionsSubset class have made this obsolete.

Future changes in this stack will further pare down code in these areas that has outlived its usefulness

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by OwenKephart** [comment]
* **#25812** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25812?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/time_window_partitions.py [hunk 1]
```diff
@@ -1687,24 +1685,140 @@ def inner(
...
+    included partitions using TimeWindows.
     """
 
     # Every time we change the serialization format, we should increment the version number.
     # This will ensure that we can gracefully degrade when deserializing old data.
     SERIALIZATION_VERSION = 1
 
-    @abstractproperty
-    def included_time_windows(self) -> Sequence[PersistedTimeWindow]: ...
+    def __new__(
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/_core/definitions/time_window_partitions.py:1725]: This has confused the diff algorithm, but I solemnly swear that all I've done in this file is:

- Delete PartitionKeysTimeWindowPartitionsSubset
- Move all methods from BaseTimeWindowPartitionsSubset to TimeWindowPartitionsSubset
- Delete BaseTimeWindowPartitionsSubset

**Comment by gibsondan** [comment]
are we confident this isn't persisted anywhere? Looks like it isn't serdes-ed so probably not?

**Comment by OwenKephart** [comment]
Yeah quite confident -- this was designed to be a purely in-memory format

**Review by clairelin135 — APPROVED** [review]
Looks good to me! Goodbye `PartitionKeysTimeWindowPartitionsSubset` 🫡

**Pull request merged**